### PR TITLE
Fix some MDX 2.4 revdep failures

### PR DESCRIPTION
**Pull request by Leonidas-from-XIV**
While investigating possible regressions in dune 3.4.1 I ran into some new failures that I have tracked down to be not caused by dune but rather MDX 2.4.0: #25355 so this PR fixes some of the most obvious ones.

### CODE REVIEW

**Comment by Leonidas-from-XIV** [comment]
There's a bunch more broken test cases that I saw, in `grpc-examples`, `irmin-cli` and `pretty_expressive` if a maintainer wishes to fix them.

**Comment by mseri** [comment]
Thanks

File: packages/eio_main/eio_main.0.15/opam [hunk 1]
```diff
@@ -9,7 +9,7 @@ doc: "https://ocaml-multicore.github.io/eio/"
 bug-reports: "https://github.com/ocaml-multicore/eio/issues"
 depends: [
   "dune" {>= "3.9"}
-  "mdx" {>= "2.2.0" & with-test}
+  "mdx" {>= "2.2.0" & < "2.4" & with-test}
   "kcas" {>= "0.3.0" & with-test}
```
Comment (mseri) [inline packages/eio_main/eio_main.0.15/opam:13]:
```suggestion
  "kcas" {>= "0.3.0" & < "0.7.0" & with-test}
```

Comment (mseri) [reply packages/eio_main/eio_main.0.15/opam:13]: Same fix as in the previous version

**Comment by mseri** [comment]
Thanks!

**Pull request merged**